libyara/modules/dex.c

```diff
@@ -0,0 +1,743 @@
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <yara/modules.h>
+#include <yara/mem.h>
+#include <yara/endian.h>
+
+#define IMAGE_DEX_SIGNATURE_035 (uint8_t[8]) { 0x64, 0x65, 0x78, 0x0A, 0x30, 0x33, 0x35, 0x00 }
+#define IMAGE_DEX_SIGNATURE_036 (uint8_t[8]) { 0x64, 0x65, 0x78, 0x0A, 0x30, 0x33, 0x36, 0x00 }
+#define IMAGE_DEX_SIGNATURE_037 (uint8_t[8]) { 0x64, 0x65, 0x78, 0x0A, 0x30, 0x33, 0x37, 0x00 }
+#define IMAGE_ODEX_SIGNATURE_035 (uint8_t[8]) { 0x64, 0x65, 0x79, 0x0A, 0x30, 0x33, 0x35, 0x00 }
+#define IMAGE_ODEX_SIGNATURE_036 (uint8_t[8]) { 0x64, 0x65, 0x79, 0x0A, 0x30, 0x33, 0x36, 0x00 }
+#define IMAGE_ODEX_SIGNATURE_037 (uint8_t[8]) { 0x64, 0x65, 0x79, 0x0A, 0x30, 0x33, 0x37, 0x00 }
+#define ENDIAN_CONSTANT 0x12345678
+#define REVERSE_ENDIAN_CONSTANT 0x78563412
+
+#define member_size(type, member) sizeof(((type *)0)->member)
+
+#define MODULE_NAME dex
+
+typedef struct {
```

Comment:
Why all fields in these structs are 1-sized arrays? O_o 

This leads to a lot of confusing expressions like `*dex_header->string_ids_size` where they should be simply `dex_header->string_ids_size`. In fact I was wondering how `*dex_header->string_ids_size` compiled at all if `dex_header` was a pointer to a struct, until I discovered all these 1-element arrays.